app/components/license/UploadLicense.tsx

```diff
@@ -1,12 +1,28 @@
-// @ts-nocheck File not migrated fully to TS
 import i18n from 'i18next';
 import _ from 'lodash';
-import PropTypes from 'prop-types';
 import React from 'react';
+import type { ChangeEvent } from 'react';
+import type { ButtonProps, FormProps } from 'semantic-ui-react';
 
-import { Button, ErrorMessage, Form, Segment } from '../basic';
+import { Button, Form, Segment } from '../basic';
 
-export default function UploadLicense({ error, isLoading, license, onChange, onErrorDismiss, onUpload }) {
+interface UploadLicenseProps {
+    error?: string;
+    isLoading: boolean;
+    license: string;
+    onChange: (event: ChangeEvent<HTMLTextAreaElement>, data: any) => void;
```

Comment:
Maybe we can have here:
```suggestion
    onChange: StrictTextAreaProps['onChange'];
```
? 🤔

Reply:
Changing it causing:
```
Issues checking in progress...
ERROR in app/components/LicensePage.tsx:296:29
TS2322: Type '(_event: ChangeEvent<HTMLTextAreaElement>, field: { name: string; value: unknown; type: string; checked?: boolean | undefined; }) => void' is not assignable to type '(event: ChangeEvent<HTMLTextAreaElement>, data: TextAreaProps) => void'.
  Types of parameters 'field' and 'data' are incompatible.
    Type 'TextAreaProps' is missing the following properties from type '{ name: string; value: unknown; type: string; checked?: boolean | undefined; }': name, type
    294 |                             isLoading={isLoading}
    295 |                             license={licenseString}
  > 296 |                             onChange={this.onLicenseEdit}
        |                             ^^^^^^^^
    297 |                             onErrorDismiss={this.onErrorDismiss}
    298 |                             onUpload={this.onLicenseUpload}
    299 |                         />

```